contrib/testing/integrations.sh

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash
+# Copyright 2021 Authors of Cilium
+# SPDX-License-Identifier: Apache-2.0
+
+# Ginkgo Cilium Test
+#
+# $1 - integration
+# $2 - cilium image
+# $3 - focus string
+# $4 - additional ginkgo arguments
+gct()
+{
+    if [ $# -lt 2 ]; then
+        >&2 echo "usage: gct <INTEGRATION> <IMAGE> [FOCUS] [GINKGO-ARGS]"
+        return 1
+    fi
+    for dep in jq kubectl ginkgo; do
+        if ! which $dep >/dev/null; then
+            >&2 echo "This script requires '$dep'"
+            return 1
+        fi
+    done
+    (
+        set -e
+        set -o pipefail
+
+        if [ ! -e $PWD/test_suite_test.go ]; then
+            CILIUM_DIR="${CILIUM_DIR:-"$GOPATH/src/github.com/cilium/cilium/test"}"
+            >&2 echo "Switching to $CILIUM_DIR..."
+            cd "$CILIUM_DIR"
+        fi
+        local k8s_version=$(kubectl version -o json)
+        local k8s_major=$(echo "$k8s_version" | jq '.serverVersion.major')
+        local k8s_minor=$(echo "$k8s_version" | jq '.serverVersion.minor')
+
+        CNI_INTEGRATION="$1"; shift
+        CILIUM_IMAGE="$(echo "$1" | sed 's/^\(.*\):[^:]*$/\1/')"
+        CILIUM_TAG="$(echo "$1" | sed 's/^.*:\([^:]*\)$/\1/')"
+        shift
+        FOCUS=""
+        if [ $# -ge 1 ]; then
+            FOCUS="--focus=$1"
+        fi
+        shift
+        CNI_INTEGRATION="$CNI_INTEGRATION" \
+        CILIUM_IMAGE="$CILIUM_IMAGE" \
+        CILIUM_TAG="$CILIUM_TAG" \
+        CILIUM_OPERATOR_IMAGE="${CILIUM_OPERATOR_IMAGE:-"docker.io/cilium/operator"}" \
+        CILIUM_OPERATOR_TAG="${CILIUM_OPERATOR_TAG:-"latest"}" \
+        K8S_VERSION="$k8s_major.$ks_minor" \
+        ginkgo -v "$FOCUS" -- \
+            -cilium.provision=false \
+            -cilium.kubeconfig=$HOME/.kube/config \
+            -cilium.passCLIEnvironment=true \
+            -cilium.testScope=k8s \
+            -cilium.holdEnvironment=true \
+            -cilium.skipLogs=true \
+            "$@"
+    )
+}
+
+# Ginkgo for eKS
+gks()
+{
+    if [ $# -lt 2 ]; then
+        >&2 echo "usage: gks <IMAGE> [FOCUS] [ARGS]"
+        return 1
+    fi
+    IMAGE="$1"; shift
+    FOCUS="$2"; shift
```

Comment:
I may be missing something here, but shouldn't this be `FOCUS="$1"; shift` since we already shifted in the line above?

I tested it locally with following script:

```
gks()    
{
    if [ $# -lt 2 ]; then
        >&2 echo "usage: gks <IMAGE> [FOCUS] [ARGS]"
        exit 1
    fi
    IMAGE="$1"; shift
    FOCUS="$2"; shift

	echo $IMAGE $FOCUS
}

gks trolo lolo
```

and I am getting correct output only if I set `FOCUS` to `"$1"`

Reply:
Good spotting, will fix.